step3/dataStores/Storage.js

```diff
@@ -0,0 +1,46 @@
+const uuidv4 = require('uuid/v4');
+
+class Storage {
+
+	constructor () {
+			this.messages = [];
+			this.users = [];
+	}
+
+	saveMessage(username, body) {
+		this.messages.push({
+			username: username,
+			body: body,
+			timestamp: Date.now(),
+			id: uuidv4()});
+	}
+
+	getMessages() {
+		return this.messages;
+	}
+
+	connect(username) {
+		const index = this.users.indexOf(username);
```

Comment:
this.users.includes(username)

Reply:
https://nodesource.com/blog/six-of-the-most-exciting-es6-features-in-node-js-v6-lts/